src/github.com/matrix-org/dendrite/clientapi/writers/sendmessage.go

```diff
@@ -14,7 +15,9 @@ type SendMessage struct {
 // OnIncomingRequest implements util.JSONRequestHandler
 func (s *SendMessage) OnIncomingRequest(req *http.Request) (interface{}, *util.HTTPError) {
 	logger := req.Context().Value(util.CtxValueLogger).(*log.Entry)
-	logger.Info("Doing stuff...")
+	vars := mux.Vars(req)
+	roomID := vars["roomID"]
+	logger.WithField("roomID", roomID).Info("Doing stuff...")
```

Comment:
The implementation of `mux.Vars` is "interesting". There should probably be a comment saying that this handler has to be called through a gorilla Mux in order for it to have stashed the route vars on the request context.